packages/components/addon/components/hds/tabs/index.js

```diff
@@ -56,6 +56,11 @@ export default class HdsTabsIndexComponent extends Component {
       block: 'nearest',
       inline: 'nearest',
     });
+
+    if (typeof this.args.onTabClick === 'function') {
+      const tabElem = this.tabNodes[tabIndex];
+      this.args.onTabClick(tabElem, ...arguments);
```

Comment:
As "consumer" I would expect that the arguments provided by this callback are exactly the same as the standard `onClick` DOM event handler, so I would do something simpler that doesn't require explanations or documentation. It's just a `onClick` propagated to the `Tab` (and if they want to get the element, they can always call `event.target`).
Something like this:

```
@action
handleClick(tabIndex, event) {
  ...
  if (typeof this.args.onTabClick === 'function') {
    this.args.onTabClick(event);
  }
}
```